### PR TITLE
[codex] Prototype image gallery controls

### LINKED ISSUE #7412: Prototype image gallery controls and RF card handoff
## Intent
Prototype the image gallery interaction updates requested from this device.

## Requirements
- Show image information on the left and the card/image on the right.
- Add gallery search across image field values.
- Add previous/next navigation buttons.
- Include the image ID in feedback context that is saved and copied to clipboard.
- Add a Use for RF Card link when a compatible store is set up.

## Verification
- .venv/bin/python manage.py test run -- apps/gallery/tests/test_gallery.py apps/shop/tests/test_checkout.py apps/sites/tests/test_user_story_feedback_form.py

### PR DESCRIPTION
**Pull request by arthexis**
## Summary
- reorganizes gallery detail pages with metadata on the left and the image/card preview on the right
- adds gallery search across image field values and related metadata plus previous/next navigation
- carries image ID context into feedback clipboard/submission payloads and links compatible gallery images into RF card store customization

## Issue
Closes #7412

## Verification
- .venv/bin/python manage.py test run -- apps/gallery/tests/test_gallery.py apps/shop/tests/test_checkout.py apps/sites/tests/test_user_story_feedback_form.py
- git diff --check

## Notes
- black and ruff are not installed in this virtualenv, so formatting was kept manual

### Overview
Reorganizes gallery detail pages into a two-column layout (metadata left, image/card preview right), adds multi-field gallery search with refined typed matching and viewer-scoped metadata gates, replaces list-based previous/next navigation with DB-level adjacent queries ordered by (title, id) for deterministic tie-breaking, carries image ID context into feedback clipboard/submission payloads, and surfaces a conditional "Use for RF Card" handoff when a compatible store/product exists.

### Gallery Enhancements

Search & Query Behavior
- Adds search via the q parameter matching direct image fields (id, UUID/slug/content-sample-name, title, description, filename, include_in_public_gallery) and related metadata (categories, traits, credits).
- Typed parsing: integer queries match exact image id; UUID-like strings match exact gallery slug or content-sample name; boolean-like strings map to include_in_public_gallery; float parsing applied for numeric trait values; text fields remain substring-matching.
- Splits search into direct-field and related-field subqueries to isolate multi-valued joins; final queryset composes results by combining PK subqueries via id__in to avoid join blowup.
- Metadata-only search predicates are gated by per-viewer can_view_metadata so viewers without metadata access cannot match or surface owner/owner-group, category/trait, credit, or content-sample internals.

Navigation & Ordering
- Replaces in-memory list navigation with DB-level adjacent queries ordered by (title, id) to ensure deterministic same-title tie-breaking.
- Previous/Next and Back links preserve q/search context when present.

Detail Page Reorganization
- Two-column responsive layout with a header containing Back/Previous/Next controls and a conditional "Use for RF Card" link when rf_card_store_url is available.
- Left column lists attributes (id/slug, filename/original name, public/private visibility); owner, owner-group, content-sample name, credits, categories, and traits are rendered only when can_view_metadata permits; main image/preview and description moved to the right.

Prefetch & Performance
- Removes an overly broad prefetch_related() from _visible_images_for_user(); retains targeted prefetches where required by rendering.
- Adds test assertions verifying multi-valued join filters remain constrained to PK subqueries and do not appear on the outer gallery queryset.

Metadata Visibility & Privacy
- Preserves GalleryImage.can_view_metadata() semantics; owner/owner-group, content-sample internals, and metadata-only search predicates are hidden from viewers lacking metadata permission.
- Tests confirm shared recipients can view images without seeing owner identity; owners retain full metadata visibility.

RF Card Store Integration & Shop Handoff
- Detail view computes rf_card_store_url when a compatible, active, in-stock ShopProduct supports gallery-image printing; detail page shows "Use for RF Card" link if present.
- shop_index accepts a gallery_image query param, validates visibility for checkout, and stores a pending session handoff when valid.
- add_to_cart validates posted gallery_image for qualifying products, persists {product_id, gallery_image_id} in session when valid, preserves pending handoff across unrelated/non-customizable adds, applies pending handoff to the first matching customizable add, and clears stale handoff when a customizable product is added without a fresh valid gallery_image or when the stored selection becomes invalid.
- checkout pre-fills the stored gallery image for the matching customization entry and clears session handoff after order placement.

Feedback Context Tracking
- Gallery detail embeds image ID into data-feedback-context; client JS deduplicates and syncs context strings into input[name="feedback_context"] (1000-char limit).
- UserStoryForm adds an optional hidden feedback_context field, cleaner, and logic to append a formatted "Context: ..." suffix to messages without duplicating an existing context line; stored messages are truncated to 2000 chars.
- JS copy behavior excludes feedback_context from copied staff-detail fields to avoid duplicate context lines while preserving hidden-field submit-time persistence.
- Tests verify copied clipboard text contains a single Image ID entry and that hidden feedback_context syncs for submission.

Test Coverage & Verification
- Adds focused gallery tests covering index/detail search semantics (direct vs related), exact-match numeric and UUID behaviors, navigation tie-breaker ordering, metadata visibility gating, and query-shape assertions preventing join blowup.
- Adds shop checkout tests exercising gallery-image handoff propagation, persistence across cart mutations, unrelated-customizable preservation, and stale-handoff clearing.
- Adds a sites form test asserting feedback_context is appended to stored messages.
- Author reports iterative focused test runs and basic static checks during development.

Files of Note
- apps/gallery/views.py: search implementation, adjacent navigation queries, metadata visibility, rf_card_store_url computation.
- apps/gallery/templates/gallery/detail.html & index.html: two-column detail layout, search UI, preserved search-context navigation, RF card link.
- apps/gallery/tests/test_gallery.py: expanded tests for search semantics, navigation, metadata visibility, and query-shape.
- apps/shop/views.py, templates, apps/shop/tests/test_checkout.py: gallery-image handoff validation, session persistence, and checkout preselection tests.
- apps/sites/forms.py, apps/sites/static/pages/js/user_story_feedback.js, templates, and apps/sites/tests/test_user_story_feedback_form.py: feedback_context handling, client-side sync, copy behavior, and form tests.

Closes: Issue #7412

### CODE REVIEW

**Comment by arthexis** [comment]
Worker update for the unresolved Gemini review threads.

Changed:
- Replaced list-based detail previous/next navigation with DB-level adjacent queries ordered by `(title, id)`, including the same-title tie-breaker.
- Removed the broad `prefetch_related()` from `_visible_images_for_user()` after checking call sites; detail and metadata views keep their targeted prefetches where rendered metadata needs them.
- Removed cast-based gallery search annotations. Numeric queries now match exact `id`; UUID queries match exact gallery `slug` or content sample `name`; existing direct text-field searches remain unchanged.
- Added focused gallery tests for exact id search, exact slug search, and same-title navigation ordering.

Verification in container `quarry-pr7413`:
- `python manage.py test apps.gallery.tests.test_gallery` attempted; this repo wrapper requires an action/path form, so the equivalent runnable command was used.
- `.venv/bin/python manage.py test run apps/gallery/tests/test_gallery.py` -> 21 passed.
- `git diff --check` -> passed.
- `.venv/bin/python -m compileall apps/gallery/views.py apps/gallery/tests/test_gallery.py` -> passed.
- `.venv/bin/ruff check apps/gallery/views.py apps/gallery/tests/test_gallery.py` -> passed.

Skipped findings: none. Scope stayed limited to `apps/gallery/views.py` and focused gallery tests.

File: apps/gallery/views.py [hunk 1]
```diff
@@ -72,9 +75,106 @@ def _visible_images_for_user(user):
...
+    if search_query:
+        filtered_queryset = _apply_gallery_search(queryset, search_query)
+        if filtered_queryset.filter(pk=image.pk).exists():
+            queryset = filtered_queryset
+    images = list(queryset.order_by("title", "id"))
+    current_index = next((index for index, candidate in enumerate(images) if candidate.pk == image.pk), None)
+    if current_index is None:
+        return None, None
+    previous_image = images[current_index - 1] if current_index > 0 else None
+    next_image = images[current_index + 1] if current_index < len(images) - 1 else None
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/gallery/views.py]: Addressed in commit 969441cd: detail previous/next navigation now uses database-level adjacent queries ordered by `(title, id)` instead of materializing the queryset. Worker verification reported gallery tests passing plus compile, Ruff, and diff checks.

File: apps/gallery/views.py [hunk 2]
```diff
@@ -61,7 +64,7 @@ def _visible_images_for_user(user):
         "media_file",
         "owner_user",
         "owner_group",
-    )
+    ).prefetch_related("categories", "credits", "trait_values__category", "trait_values__trait")
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/gallery/views.py]: Addressed in commit 969441cd: the broad base queryset prefetch was removed after call-site review, while detail and metadata views keep targeted prefetches where rendered metadata needs them. Worker verification reported gallery tests passing plus compile, Ruff, and diff checks.

File: apps/gallery/views.py [hunk 3]
```diff
@@ -72,9 +75,106 @@ def _visible_images_for_user(user):
...
 
+def _apply_gallery_search(queryset, search_query: str):
+    search_query = search_query.strip()
+    if not search_query:
+        return queryset
+
+    queryset = queryset.annotate(
+        search_id=Cast("id", output_field=CharField()),
+        search_slug=Cast("slug", output_field=CharField()),
+        search_content_sample_name=Cast("content_sample__name", output_field=CharField()),
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/gallery/views.py]: Addressed in commit 969441cd: gallery search no longer casts id/slug for `icontains`; numeric ids and UUID slugs/content sample names use exact matching while direct text searches remain unchanged. Worker verification reported gallery tests passing plus compile, Ruff, and diff checks.

**Comment by arthexis** [comment]
Worker follow-up for CodeRabbit thread `PRRT_kwDOPUOpW859lEHH`.

Patch: `da52ee6500c0d96e771b9ccdc9cf7a89205e8426`

What changed:
- Split `_apply_gallery_search` into `direct_fields_q` for scalar/single-valued fields and `related_fields_q` for multi-valued category, credit, and trait matches.
- The final gallery queryset now combines two PK subqueries with `id__in` instead of applying every relation traversal in one broad outer `Q`, so the expensive multi-valued joins stay isolated inside the related-match subquery.
- Preserved existing search behavior for text fields, exact numeric id, UUID slug/content-sample name, public/private aliases, and exact trait float matching.
- Added gallery tests covering direct plus multi-valued matches and asserting the outer gallery query does not carry the multi-valued joins.

Verification run inside container `silt-pr7413`:
- `.venv/bin/python manage.py test run apps/gallery/tests/test_gallery.py` -> `23 passed`
- `git diff --check`
- `.venv/bin/python -m compileall -q apps/gallery/views.py apps/gallery/tests/test_gallery.py`
- `.venv/bin/ruff check apps/gallery/views.py apps/gallery/tests/test_gallery.py`

Residual risk: this is still substring search across several fields and it is still unpaginated; this patch specifically bounds the join blowup called out in the blocking review without introducing pagination or full-text search in the prototype PR.

**Comment by arthexis** [comment]
Addressed the Codex P1 owner metadata disclosure thread.

Patch:
- gated the gallery detail owner/owner group rows behind `can_view_metadata`, preserving the `GalleryImage.can_view_metadata()` boundary for users who can view an image only through sharing
- added focused shared-image detail tests confirming a shared recipient can view the image without seeing owner identity, while the owner still sees owner metadata

Container verification:
- `.venv/bin/python manage.py test run apps/gallery/tests/test_gallery.py` (`25 passed`)
- `git diff --check`
- `.venv/bin/python -m compileall apps/gallery`
- `.venv/bin/ruff check apps/gallery/tests/test_gallery.py`
- `.venv/bin/python manage.py check`

I did not merge, approve, mark ready, resolve threads, close the PR, or change labels/priority.

**Comment by arthexis** [comment]
Worker update for the feedback copy payload issue.

Changed:
- Added `feedback_context` to the copy-details FormData skip list in `apps/sites/static/pages/js/user_story_feedback.js`.
- Kept submit-time FormData unchanged, so the hidden `feedback_context` field still syncs and persists through `UserStoryForm`.

Verification, all in container `tackle-pr7413`:
- `NODE_PATH=/tmp/tackle-jsprobe/node_modules node <jsdom copy probe>`: passed; copied text contains `Image ID: 42` once, contains no literal `feedback_context:`, and hidden `feedback_context` syncs to `Image ID: 42 | Image UUID: abc`.
- `.venv/bin/python -m pytest apps/sites/tests/test_user_story_feedback_form.py -q`: `2 passed`.
- `node --check apps/sites/static/pages/js/user_story_feedback.js`: passed.
- `.venv/bin/python -m compileall -q apps/sites`: passed.
- `git diff --check`: passed.

Note: `./env-refresh.sh --deps-only` hit a PyPI read timeout while downloading the large CI `opencv-python` wheel after base Django dependencies were installed; I installed only the focused pytest dependencies needed for the sites feedback test afterward. No gallery/shop files were touched.

**Comment by arthexis** [comment]
Worker update for the replacement gallery/shop scope.

Changed:
- Gated gallery metadata-only search predicates behind per-viewer metadata visibility, so shared/public viewers can still search public image fields and image IDs/slugs but cannot match owner identity, owner group, category/trait, credits, or content-sample internals unless they can view metadata for that image.
- Hid the content sample name on gallery detail behind the same `can_view_metadata` boundary as owner metadata.
- Cleared stale `shop_cart_gallery_image` session handoff when a customizable product is added without a fresh `gallery_image` post value, and also clears it for invalid posted selections.
- Added focused regressions for shared/public metadata search suppression, owner metadata search, detail content-sample disclosure, and stale shop handoff clearing.

Verification in container `mosaic-pr7413`:
- `.venv/bin/python manage.py test run apps/gallery/tests/test_gallery.py apps/shop/tests/test_checkout.py` -> 53 passed
- `.venv/bin/python -m compileall apps/gallery/views.py apps/shop/views.py apps/gallery/tests/test_gallery.py apps/shop/tests/test_checkout.py`
- `.venv/bin/ruff check apps/gallery/views.py apps/gallery/tests/test_gallery.py apps/shop/views.py apps/shop/tests/test_checkout.py`
- `git diff --check`

No skipped edge cases in the assigned scope. I did not touch the sites feedback JS and did not merge, approve, mark ready, resolve threads, change labels, or add/remove reactions.

**Comment by arthexis** [comment]
Worker update for fresh Codex P2 thread `PRRT_kwDOPUOpW859lZMQ`.

Changed `apps/sites/static/pages/js/user_story_feedback.js` so feedback context collection and hidden `feedback_context` sync still run for non-staff copy, but staff-only copy details now live behind an explicit `canCopyStaffDetails` branch:
- `getAdminDashboardNextTask()`
- `getAdminDashboardNetMessage()`
- `getRoleSiteNodeSummary()`
- `securityGroups`

Verification run inside Docker container `quill-pr7413`:
- jsdom feedback copy probe covering non-staff context copy without staff detail leakage, staff copy unchanged, and hidden `feedback_context` de-duplication
- `.venv/bin/python manage.py test run -- apps/sites/tests/test_user_story_feedback_form.py` (`2 passed`)
- `node --check apps/sites/static/pages/js/user_story_feedback.js`
- `.venv/bin/python -m compileall apps/sites`
- `git diff --check`

I did not touch shop/gallery files and did not merge, approve, resolve threads, close issues/PRs, change labels/priority, or add/remove reactions.

**Comment by arthexis** [comment]
Worker Sable update for Codex P2 thread PRRT_kwDOPUOpW859lZMO.

Changes:
- Persist a validated /shop/?gallery_image=... handoff as pending session state before any product is added.
- Leave that pending handoff intact when unrelated/non-customizable products are added, then attach it to the first gallery-printable product added without an explicit gallery image.
- Keep clearing stale handoff state for the matched product or invalid pending gallery image while preserving unrelated product-specific handoffs.
- Added focused checkout/cart regressions for non-customizable-first flow, unrelated customizable add preservation, and stale pending cleanup.

Verification, all inside Docker container sable-pr7413:
- .venv/bin/python manage.py test run apps/shop/tests/test_checkout.py (29 passed)
- .venv/bin/python -m compileall apps/shop/views.py apps/shop/tests/test_checkout.py
- .venv/bin/ruff check apps/shop/views.py apps/shop/tests/test_checkout.py
- git diff --check

Pushed commit bf5465e73 to image-gallery-prototype-controls. I did not merge, approve, resolve threads, close PRs/issues, change labels/priority, or add/remove reactions.

File: apps/gallery/templates/gallery/detail.html [hunk 4]
```diff
@@ -4,44 +4,95 @@
...
+        <dd class="col-sm-8 text-break">{{ image.slug }}</dd>
+        <dt class="col-sm-4">{% trans "File" %}</dt>
+        <dd class="col-sm-8 text-break">{{ image.media_file.original_name|default:image.media_file.file.name }}</dd>
+        <dt class="col-sm-4">{% trans "Visibility" %}</dt>
+        <dd class="col-sm-8">
+          {% if image.include_in_public_gallery %}{% trans "Public" %}{% else %}{% trans "Private" %}{% endif %}
+        </dd>
+        {% if image.owner_user %}
+          <dt class="col-sm-4">{% trans "Owner" %}</dt>
+          <dd class="col-sm-8">{{ image.owner_user.get_username }}</dd>
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/gallery/templates/gallery/detail.html]: Addressed by commit 4e086c906: owner and owner-group rows are now behind `can_view_metadata`, with shared-viewer regression coverage reported by the worker.

File: apps/gallery/views.py [hunk 5]
```diff
@@ -72,9 +74,117 @@ def _visible_images_for_user(user):
...
+        return queryset
+
+    direct_fields_q = (
+        Q(title__icontains=search_query)
+        | Q(description__icontains=search_query)
+        | Q(owner_user__username__icontains=search_query)
+        | Q(owner_user__email__icontains=search_query)
+        | Q(owner_user__first_name__icontains=search_query)
+        | Q(owner_user__last_name__icontains=search_query)
+        | Q(owner_group__name__icontains=search_query)
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/gallery/views.py]: Addressed by commit a9df8447: metadata-only search predicates are gated by per-viewer metadata visibility, with focused shared/public search regressions reported by the worker.

File: apps/shop/views.py [hunk 6]
```diff
@@ -220,6 +236,20 @@ def add_to_cart(request: HttpRequest, shop_slug: str, product_id: int) -> HttpRe
         messages.error(request, str(exc))
     else:
         _save_cart(request, cart)
+        selected_gallery_image_id = (request.POST.get("gallery_image") or "").strip()
+        if selected_gallery_image_id and product.supports_gallery_image_printing:
+            try:
+                gallery_image_id = int(selected_gallery_image_id)
+            except ValueError:
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/shop/views.py]: Addressed by commit a9df8447: stale shop gallery handoff state is cleared for customizable products without a fresh valid posted image, with checkout regressions reported by the worker.

File: apps/gallery/templates/gallery/detail.html [hunk 7]
```diff
@@ -4,44 +4,97 @@
...
+            <dt class="col-sm-4">{% trans "Owner" %}</dt>
+            <dd class="col-sm-8">{{ image.owner_user.get_username }}</dd>
+          {% elif image.owner_group %}
+            <dt class="col-sm-4">{% trans "Owner group" %}</dt>
+            <dd class="col-sm-8">{{ image.owner_group.name }}</dd>
+          {% endif %}
+        {% endif %}
+        {% if image.content_sample %}
+          <dt class="col-sm-4">{% trans "Content sample" %}</dt>
+          <dd class="col-sm-8 text-break">{{ image.content_sample.name }}</dd>
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/gallery/templates/gallery/detail.html]: Addressed by commit a9df8447: content sample detail output is now gated by `can_view_metadata`, with detail disclosure regression coverage reported by the worker.

File: apps/shop/views.py [hunk 8]
```diff
@@ -220,6 +236,26 @@ def add_to_cart(request: HttpRequest, shop_slug: str, product_id: int) -> HttpRe
...
+                        "gallery_image_id": str(gallery_image_id),
+                    }
+                    request.session.modified = True
+                else:
+                    request.session.pop("shop_cart_gallery_image", None)
+                    request.session.modified = True
+                    messages.error(request, "Selected gallery image is unavailable for RF card customization.")
+            else:
+                request.session.pop("shop_cart_gallery_image", None)
+                request.session.modified = True
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/shop/views.py]: Addressed by commit 977762ed and superseded by bf5465e73: omitted `gallery_image` no longer erases the saved selection for the repeated-add flow; latest worker coverage also preserves pending handoff until a gallery-printable product uses it.

File: apps/shop/views.py [hunk 9]
```diff
@@ -220,6 +237,40 @@ def add_to_cart(request: HttpRequest, shop_slug: str, product_id: int) -> HttpRe
         messages.error(request, str(exc))
     else:
         _save_cart(request, cart)
+        selected_gallery_image_id = (request.POST.get("gallery_image") or "").strip()
+        if product.supports_gallery_image_printing:
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/shop/views.py:271]: Addressed by commit bf5465e73: `/shop/?gallery_image=...` is persisted as pending session state before any product add, survives a non-customizable first add, and is attached to the first compatible product; worker reported 29 focused shop tests passing in container `sable-pr7413`.

File: apps/sites/static/pages/js/user_story_feedback.js [hunk 10]
```diff
@@ -490,19 +492,43 @@
...
+    contextField.value = contexts.join(' | ').substring(0, 1000);
+  };
+
   const buildCopyValue = () => {
     const baseValue = getPageCopyValue();
-    if (!canCopyStaffDetails) {
+    const feedbackContexts = getFeedbackContextLines();
+    syncContextField(feedbackContexts);
+    if (!canCopyStaffDetails && !feedbackContexts.length) {
       return baseValue;
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/sites/static/pages/js/user_story_feedback.js:526]: Addressed by commit 9d0714c8: feedback context sync still runs, but staff-only copy helpers and security groups are explicitly gated by `canCopyStaffDetails`; worker reported jsdom and focused form tests passing in container `quill-pr7413`.

**Comment by arthexis** [comment]
Worker Opal update for PR #7413 review blockers.

Commit: `15311ddff` (`Address gallery visibility search review blockers`)

Changed files:
- `apps/gallery/views.py`
- `apps/gallery/tests/test_gallery.py`
- `apps/shop/templates/shop/index.html`

Patch summary:
- Hoisted `ShopProduct` import to module scope in `apps/gallery/views.py`; current branch state showed no circular import blocker.
- Tightened gallery visibility keyword search so common words `yes`, `no`, `true`, and `false` no longer expand to public/private bulk visibility matches. Explicit `public`, `published`, and `private` visibility searches remain supported.
- Added a focused gallery regression covering the common boolean word behavior.
- Replaced the hardcoded inline `style="width: 78px;"` quantity wrapper in the shop index template with repository/bootstrap class sizing (`col-auto`) and no fixed inline px/rem sizing.

Container verification (`opal-pr7413`, `/work/arthexis`):
- `./env-refresh.sh --deps-only` completed inside the container.
- `.venv/bin/python manage.py test run -- apps/gallery/tests/test_gallery.py apps/shop/tests/test_checkout.py` -> `58 passed in 115.64s`.
- `.venv/bin/python -m compileall apps/gallery apps/shop` -> passed.
- `.venv/bin/ruff check apps/gallery/views.py apps/gallery/tests/test_gallery.py` -> `All checks passed!`.
- `git diff --check` -> passed.

Skipped feedback:
- No feedback in the assigned three unresolved threads was skipped.
- I did not address unrelated PR #7413 threads or broader gallery/shop refactors outside the assigned scope.
- I did not merge, approve, mark ready, resolve review threads, close issues/PRs, change labels/priority, or add/remove reactions.

File: apps/shop/templates/shop/index.html [hunk 11]
```diff
@@ -21,10 +26,16 @@ <h3 class="h5">{{ product.name }}</h3>
...
+                    {% endif %}
                     <div class="d-flex justify-content-between align-items-end gap-2 mb-2">
                       <div>
                         <p class="mb-1"><strong>{{ product.unit_price }} {{ product.currency }}</strong></p>
                         <p class="text-muted small mb-0">Stock: {{ product.stock_quantity }}</p>
+                        {% if selected_gallery_image and product.supports_gallery_image_printing %}
+                          <p class="text-muted small mb-0">RF card image: {{ selected_gallery_image.title }}</p>
+                        {% endif %}
                       </div>
                       <div style="width: 78px;">
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/shop/templates/shop/index.html]: Addressed in 15311ddff: the hardcoded inline width was removed and the quantity wrapper now uses repository/bootstrap class sizing without fixed px/rem style.

**Comment by arthexis** [comment]
Worker patch for fresh Codex P2 thread `PRRT_kwDOPUOpW859lwzT`.

Summary:
- Updated `apps/shop/views.py` so `/shop/` with no `gallery_image` query param validates existing `shop_cart_gallery_image` session state.
- Rehydrates a valid stored handoff into `selected_gallery_image`, keeping the visible banner/form context in sync with the session-backed add-to-cart/checkout behavior.
- Clears malformed or stale stored handoff state when the referenced gallery image is no longer visible/available.
- Added focused checkout regressions for valid pending handoff rehydration and stale pending handoff clearing.

Verification inside Docker container `onyx-pr7413`:
- `.venv/bin/python manage.py test run apps/shop/tests/test_checkout.py` -> `31 passed in 49.32s`
- `.venv/bin/python -m compileall apps/shop` -> passed
- `.venv/bin/ruff check apps/shop/views.py apps/shop/tests/test_checkout.py` -> passed
- `git diff --check` -> passed

Skipped feedback rationale:
- Only addressed thread `PRRT_kwDOPUOpW859lwzT`; gallery search, sites feedback JS, template design-token work, #7426, and unrelated checkout refactors were intentionally left out of scope.
- I did not merge, approve, mark ready, resolve threads, change labels/priority, close anything, or add/remove reactions.

File: apps/shop/views.py [hunk 12]
```diff
@@ -86,6 +87,28 @@ def _gallery_images_for_checkout(request: HttpRequest) -> QuerySet[GalleryImage]
     return queryset.filter(visibility_filter).distinct()
 
 
+def _selected_gallery_image_for_store(request: HttpRequest) -> GalleryImage | None:
+    selected_gallery_image_id = (request.GET.get("gallery_image") or "").strip()
+    if not selected_gallery_image_id:
+        return None
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/shop/views.py:112]: Addressed in 983b8408: /shop/ now rehydrates valid pending gallery handoff session state when gallery_image is absent and clears invalid stale state, with focused checkout regressions.

**Comment by arthexis** [comment]
Worker patch for fresh Codex P2 thread `PRRT_kwDOPUOpW859l6Lx` / review comment `3142125126`.

Summary:
- Updated `apps/shop/views.py` so `/shop/?gallery_image=` and non-numeric `gallery_image` values clear `shop_cart_gallery_image` before returning no selected image.
- This keeps the invisible UI state aligned with session state, preventing a prior pending handoff from being consumed later by add-to-cart/checkout.
- Added focused checkout/store regression coverage for empty and non-numeric query values, including a later customizable add that must not preselect the stale image at checkout.

Commit: `63fc897b3c22e90db14820af73c85ee3fd0ffbfd`

Verification inside Docker container `onyx-pr7413`:
- `.venv/bin/python manage.py test run apps/shop/tests/test_checkout.py` -> `32 passed in 50.88s`
- `.venv/bin/python -m compileall apps/shop` -> passed
- `.venv/bin/ruff check apps/shop/views.py apps/shop/tests/test_checkout.py` -> passed
- `git diff --check` -> passed

Skipped feedback rationale:
- Only addressed thread `PRRT_kwDOPUOpW859l6Lx`; gallery search, sites feedback JS, template design-token work, #7426, and broader checkout refactors were intentionally left out of scope.
- I did not merge, approve, mark ready, resolve review threads, change labels/priority, close anything, or add/remove reactions.

File: apps/shop/views.py [hunk 13]
```diff
@@ -86,6 +87,45 @@ def _gallery_images_for_checkout(request: HttpRequest) -> QuerySet[GalleryImage]
...
+            request.session.modified = True
+        return image
+
+    selected_gallery_image_id = selected_gallery_image_id.strip()
+    if not selected_gallery_image_id:
+        return None
+    try:
+        image_id = int(selected_gallery_image_id)
+    except ValueError:
+        return None
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/shop/views.py:118]: Addressed in 63fc897b: invalid or empty gallery_image query values now clear the stored pending handoff so the shop UI and later add-to-cart behavior cannot diverge, with focused checkout regressions.
